lfs/util.go

```diff
@@ -310,3 +310,35 @@ func FileExistsOfSize(path string, sz int64) bool {
 
 	return !fi.IsDir() && fi.Size() == sz
 }
+
+func CopyFileContents(src string, dst string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+	out, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		cerr := out.Close()
+		if err == nil {
+			err = cerr
+		}
+	}()
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return err
+	}
+	err = out.Sync()
+	return err
+}
+
+func LinkOrCopy(src string, dst string) error {
+	err := os.Link(src, dst)
```

Comment:
@sinbad Thanks for your feedback. About the temporary file location: do you think it would be better to use `os.TempDir()` or store the tempfile in the destination directory? Latter is probably better from performance point of view because `os.TempDir()` might be in a different file system.